notebooks/sycamore-tutorial-intermediate-etl.ipynb

```diff
@@ -0,0 +1,410 @@
+    "\n",
+    "We create the schema using the `with_property` transform and attach it to each document as a property. Then, we `extract_properties` using OpenAI's GPT. You'll need to set your OpenAI key in your env or in the Op The `num_of_elements` parameter determines how many elements from each document will be sent to the LLM for extraction."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "from sycamore.transforms.extract_schema import OpenAIPropertyExtractor\n",
+    "from sycamore.llms import OpenAI, OpenAIModels\n",
+    "\n",
+    "llm = OpenAI(OpenAIModels.GPT_4O.value)\n",
+    "\n",
+    "enriched_docset = (regex_docset\n",
+    "    .with_property('_schema_class', lambda d: 'FlightAccidentReport')\n",
+    "    .with_property('_schema', lambda d: {\n",
+    "            'type': 'object',\n",
+    "            'properties': {'accidentNumber': {'type': 'string'},\n",
+    "                           'dateAndTime': {'type': 'string'},\n",
+    "                           'location': {'type': 'string'},\n",
+    "                           'aircraft': {'type': 'string'},\n",
+    "                           'aircraftDamage': {'type': 'string'},\n",
+    "                           'injuries': {'type': 'string'},\n",
+    "                           'definingEvent': {'type': 'string'}},\n",
+    "            'required': ['accidentNumber',\n",
+    "                         'dateAndTime',\n",
+    "                         'location',\n",
+    "                         'aircraft',\n",
+    "                         'aircraftDamage',\n",
+    "                         'injuries',\n",
+    "                         'definingEvent']})\n",
+    "    .extract_properties(property_extractor=OpenAIPropertyExtractor(llm=llm, num_of_elements=35,))\n",
+    ")\n",
+    "\n",
+    "enriched_docset.show(limit=1)\n",
+    "#show the additional properties extracted and included as metadata"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## Chunking\n",
+    "\n",
+    "Chunking is the process of combining (or splitting) the elements of your documents into larger (or smaller) elements (or referred to as chunks). This is an important step before creating embeddings, which are generated for each chunk.\n",
+    "\n",
+    "Sycamore can implement a few different chunking strategies. In this example, we'll use the `MarkedMerger` which relies on \"marks\" placed on the elements to decide what elements to merge into larger chunks, and what elements to drop completely. We'll use the `mark_bbox_preset` transform to add these marks to the elements in our DocSet.\n",
+    "\n",
+    "We'll also need to incorporate some information about the embedding model we're planning on using in order to limit context lengths appropriately. After merging, we'll also split any elements that are too big using `split_elements`."
```

Comment:
Put the first sentence in the previous paragraph

Reply:
Ack